### DISCUSSION TITLE
Cluster_name is not seeded from the first node

### DISCUSSION THREAD

**Discussion by luos** [rabbitmq/rabbitmq-server#7034] [Other]
Hi,

We were investigating federation performance problems related to https://github.com/rabbitmq/rabbitmq-server/pull/3085 and were looking at the docs on https://www.rabbitmq.com/configure.html about `cluster_name`. 

The docs mention that `cluster_name` is seeded from the first node in the cluster, but it does not seem to be the case. I seem to remember it used to work like that, but maybe not. 

I tested this with RabbitMQ 3.11.6, started up 3 nodes in docker, then joined them to the cluster manually. The result is that each node has it's "own" cluster_name. It will always fall into the `not_found` case and calculate it from a local socket.

My expectation would be that the cluster name will be something static - same for all nodes in the cluster.

**Comment by michaelklishin** [comment]
It is taken from the first node unless explicitly provided.

**Comment by michaelklishin** [comment]
Obviously if you initialize three individual nodes then join them they will originally be three clusters of their own.

Use an explicit name in that case. Peer discovery should work as well.
